control-plane/api-gateway/controllers/index.go

```diff
@@ -89,3 +127,64 @@ func gatewayForSecret(o client.Object) []string {
 	}
 	return secretReferences
 }
+
+func gatewaysForHTTPRoute(o client.Object) []string {
```

Comment:
@jm96441n also added some of these in [his PR](https://github.com/hashicorp/consul-k8s/pull/2139/files#diff-79c935b5ba76d5516918c828e4f604f0c8b78fd6f0833df8897440cee3c48981). Probably worth coordinating with him to ensure feet are not stepped on.